### ISSUE TITLE
Problem with project.ino example

### ISSUE THREAD

**Issue by clemenlg** [signetica/SunRise#1]
I tried to run the Arduino example, and don't work:
`project.ino: In function 'void setup()':
project:23:3: error: 'setenv' was not declared in this scope
   setenv("TZ", "MST7MDT,M3.2.0,M11.1.0", 1)
   ^~~~~~
project.ino:23:3: note: suggested alternative: 'getenv'
   setenv("TZ", "MST7MDT,M3.2.0,M11.1.0", 1)
   ^~~~~~
   getenv
project:31:23: error: 'utcime' was not declared in this scope
   struct tm utc_tm = *utcime(&t);
                       ^~~~~~
project.ino:31:23: note: suggested alternative: 'ctime'
   struct tm utc_tm = *utcime(&t);
                       ^~~~~~
                       ctime
project.ino: In function 'void loop()':
project:68:5: error: 'sr' was not declared in this scope
     sr.calculate(latitude, longitude, t);
     ^~
project.ino:68:5: note: suggested alternative: 'sq'
     sr.calculate(latitude, longitude, t);
     ^~
     sq
project:68:18: error: 'latitude' was not declared in this scope
     sr.calculate(latitude, longitude, t);
                  ^~~~~~~~
project:68:28: error: 'longitude' was not declared in this scope
     sr.calculate(latitude, longitude, t);
                            ^~~~~~~~~
project:68:39: error: 't' was not declared in this scope
     sr.calculate(latitude, longitude, t);
                                       ^
project.ino:68:39: note: suggested alternative: 'tm'
     sr.calculate(latitude, longitude, t);
                                       ^
                                       tm
exit status 1
'setenv' was not declared in this scope`
setenv is declared in the stdlib.h header, but don't work in Arduino

**Comment by signetica** [comment]
I am not an Arduino expert, but I'll tell you what I know, which should be enough to get you going.

I think the presence of the whole POSIX environment on the Aurduino platform depends upon the compiler which comes along with the selected board.  I used the ESP8266 for the project this code was written for and the Expressif gcc environment brought all the POSIX stuff with it.

The portion of the example that you are having trouble with just calculates the offset between the time returned by now() and UTC, which is then stored in the variable utcOffset.

In the application I wrote the code for, the POSIX timezone gets entered by the user during setup into a web interface running on an ESP8266.  If you are doing something simpler, you may be able to just remove the code that calculates utcOffset from the timezone string (in the  example, "MST7MDT,M3.2.0,M11.1.0"), and enter it directly.  For example, for EST in the US it would be 5 hours, making utcOffset = 300.

In other words, just replace the block under the comment 'Arduino TimeLib UTC offset code:' by this:
time_t utcOffset = 300;  // EST

This would need adjustment during daylight savings time but I can't help more without knowing what you are trying to accomplish.  Tell me a bit more about your platform and what you want to do and I'll perhaps be able to offer a more specific solution.

**Comment by clemenlg** [comment]
Hello,
I tried this code in a Arduino Mega board, I adapted the code, but the issue was only to inform you of this problem.
Maybe, you can add a comment  and maybe it can helps to another people.
Thanks for your work.
Regards
